qp_klp/Step.py

```diff
@@ -143,22 +149,27 @@ def parse_prep_file(cls, prep_file_path):
         # convert to standard dictionary.
         return metadata.to_dict('index')
 
-    @classmethod
-    def generate_special_map(cls, results, projects):
+    def generate_special_map(self, qclient):
         # this function should be able to be tested by passing in simulated =
         # results from qclient.
 
         # trimmed files are stored by qiita_id. Find the qiita_id
         # associated with each project and ensure a subdirectory exists
         # for when it comes time to move the trimmed files.
+
         special_map = []
+        results = qclient.get("/qiita_db/artifacts/types/")
+        projects = self.pipeline.get_project_info()
         for project in projects:
             upload_path = join(results['uploads'], project['qiita_id'])
             makedirs(upload_path, exist_ok=True)
             special_map.append((project['project_name'], upload_path,
                                 project['qiita_id']))
 
-        return special_map
+        self.special_map = special_map
+
+        # for ease of debugging
+        return self.special_map
```

Comment:
if this is adding `self.special_map` do we really need the return? Also, should this be an internal method (_ generate_special_map vs. generate_special_map)?

Reply:
I like it because it makes the code in test more obvious:
_obs = step.generate_special_map(fake_client)_
but I can change it to:
_step.generate_special_map(fake_client)
obs = step.special_map_